homeassistant/helpers/integration_platform.py

```diff
@@ -138,16 +186,52 @@ async def async_process_integration_platforms(
     integration_platform = IntegrationPlatform(
         platform_name, process_job, top_level_components
     )
+    # Tell the loader that it should try to pre-load the integration
+    # for any future components that are loaded so we can reduce the
+    # amount of import executor usage.
+    async_register_preload_platform(hass, platform_name)
     integration_platforms.append(integration_platform)
-
     if not top_level_components:
         return
 
     integrations = await async_get_integrations(hass, top_level_components)
-    if futures := [
-        future
-        for comp in top_level_components
-        if (platform := _get_platform(integrations[comp], comp, platform_name))
-        and (future := hass.async_run_hass_job(process_job, hass, comp, platform))
-    ]:
+    loaded_integrations: list[Integration] = []
+    for domain, integration in integrations.items():
+        if isinstance(integration, Exception):
+            _LOGGER.exception(
+                "Error importing integration %s for %s",
+                domain,
+                platform_name,
+                exc_info=integration,
+            )
+            continue
+        loaded_integrations.append(integration)
+
+    if not loaded_integrations:
+        return
+
+    # Finally, fetch the platforms for each integration and process them.
+    # This uses the import executor in a loop. If there are a lot
+    # of integration with the integration platform to process,
+    # this could be a bottleneck.
+    futures: list[asyncio.Future[None]] = []
+    for integration in loaded_integrations:
+        if not integration.platforms_exists((platform_name,)):
+            continue
+        try:
+            platform = await integration.async_get_platform(platform_name)
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unexpected error importing %s for %s",
```

Comment:
We only get `ImportError` from `async_get_platform` ([source](https://github.com/home-assistant/core/blob/e601aa0a0b74fed70cf2afb3bf3023abb937969d/homeassistant/loader.py#L1171-L1190)). Also, if it was an actual `ImportError`, it has already been logged, should we log it here again ?

Reply:
narrowed to ImportError, downgraded to debug logging

a392e59fd4c8d1a52e876d2b176010e3c740307f